api/grpcserver/post_service.go

```diff
@@ -47,12 +49,33 @@ func (s *PostService) String() string {
 	return "PostService"
 }
 
+type PostServiceOpt func(*PostService)
+
+type PostServiceOpts struct{}
+
+func (PostServiceOpts) AllowConnections() PostServiceOpt {
+	return func(s *PostService) {
+		s.allowConnections = true
+	}
+}
+
+func (PostServiceOpts) QueryInterval(interval time.Duration) PostServiceOpt {
+	return func(s *PostService) {
+		s.queryInterval = interval
+	}
+}
```

Comment:
I don't think this is a good way to implement functional options; the type `PostServiceOpts` only exists to have methods attached. It is also weird to use:

```go
psOpts := grpcserver.PostServiceOpts{}
svc := grpcserver.NewPostService(logger, psOpts.AllowConnections(), psOpts.QueryInterval(100*time.Millisecond))
```

The object `postOpts` isn't needed after its instantiation. I'd suggest to just make it a config instead:

```suggestion
type PostServiceCfg struct {
	AllowConnections bool
	QueryInterval time.Duration
}
```

And pass that as parameter to `NewPostService`. Wdyt?

Reply:
> I don't think this is a good way to implement functional options; the type PostServiceOpts only exists to have methods attached

It serves a few purposes:
1. To avoid collisions with other options in the same pkg.
2. To give the option clear meaning and purpose and combine related options together, avoiding a mess.
I mean, what an option like `grpcserver.AllowConnections()` would even mean, and why is it specific to the PostService only? 

We already have examples of the collision problem. We have:
```go
func WithPoets(poets ...PoetService) BuilderOption {
```
and 
```go
func WithPoetServices(clients ...PoetService) NIPostBuilderOption {
```

TBH, I don't think it's weird to use. A small weirdness is that an instance is needed because Go doesn't support associated functions (a method on a type)

Reply:
The name collision indicates an issue, either the services using them should be in different packages (that would also solve the `grpcserver.AllowConnections()` naming issue) or one of the two services shouldn't have that functional option.

Does it even make sense to have a `NIPostBuilder` with different PoetServices than a `Builder`? If not why does this configuration parameter need to be set twice - do we even check that the same poet services are passed to both the `Builder` and `NIPostBuilder`?